src/api/java/com/minecolonies/api/colony/requestsystem/StandardRequestManager.java

```diff
@@ -0,0 +1,1269 @@
+import com.minecolonies.api.colony.requestsystem.resolver.IRequestResolver;
+import com.minecolonies.api.colony.requestsystem.resolver.IRequestResolverProvider;
+import com.minecolonies.api.colony.requestsystem.token.IToken;
+import com.minecolonies.api.colony.requestsystem.token.StandardToken;
+import com.minecolonies.api.util.Log;
+import net.minecraft.nbt.NBTTagCompound;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Main class of the request system.
+ * Default implementation of the IRequestManager interface.
+ * <p>
+ * Uses
+ */
+public class StandardRequestManager implements IRequestManager
+{
+
+    /**
+     * BiMap that holds unique token to provider lookup.
+     */
+    @NotNull
+    private final BiMap<IToken, IRequestResolverProvider> providerBiMap = HashBiMap.create();
+
+    /**
+     * BiMap that holds unique token to resolver lookup.
+     */
+    @NotNull
+    private final BiMap<IToken, IRequestResolver> resolverBiMap = HashBiMap.create();
+
+    /**
+     * BiMap that holds unique token to request lookup.
+     */
+    @NotNull
+    private final BiMap<IToken, IRequest> requestBiMap = HashBiMap.create();
+
+    /**
+     * Map that holds the resolvers that are linked to a given provider.
+     */
+    @NotNull
+    private final HashMap<IToken, ImmutableCollection<IToken>> providerResolverMap = new HashMap<>();
+
+    /**
+     * Map that holds the requests that are linked to a given resolver.
+     */
+    @NotNull
+    private final HashMap<IToken, Collection<IToken>> resolverRequestMap = new HashMap<>();
+
+    /**
+     * Map that holds the resolver that is linked to a given request.
+     */
+    @NotNull
+    private final HashMap<IToken, IToken> requestResolverMap = new HashMap<>();
+
+    /**
+     * Map that holds the class that resolver can resolve. Used during lookup.
+     */
+    @NotNull
+    private final HashMap<Class, Collection<IRequestResolver>> requestClassResolverMap = new HashMap<>();
+    /**
+     * The fallback resolver used to resolve directly to the player.
+     * TODO: Assign resolver once implemented.
+     */
+    @NotNull
+    private final IRequestResolver                             playerResolver          = null;
+    /**
+     * Colony of the manager.
+     */
+    @NotNull
+    private IColony colony;
+
+    public StandardRequestManager(@NotNull IColony colony)
+    {
+        this.colony = colony;
+    }
+
+    /**
+     * The colony this manager manages the requests for.
+     *
+     * @return The colony this manager manages the requests for.
+     */
+    @NotNull
+    @Override
+    public IColony getColony()
+    {
+        return colony;
+    }
+
+    /**
+     * Method used to serialize the current request system to NBT.
+     *
+     * @return The NBTData that describes the current request system
+     */
+    @Override
+    public NBTTagCompound serializeNBT()
```

Comment:
No that is a wrong assumption. Day a request is made that is delayed resolved. A machine is crafting something etc, while the restart happens. Then the request needs to know that, cause it would be entirely wastefull to just restart the request, just because we did not save its state. So yes this is needed. I am not sure what it does right now (maybe not implemented yet) but if it behaves wrongly I will fix that in the final pr.